app/src/main/java/com/stripe/aod/sampleapp/model/MainViewModel.kt

```diff
@@ -60,12 +67,34 @@ class MainViewModel : ViewModel() {
         }
     }
 
+    init {
+        viewModelScope.launch {
+            launch {
+                TerminalEventListener.onConnectionStatusChange.collect(::updateConnectStatus)
+            }
+
+            launch {
+                TerminalEventListener.onPaymentStatusChange.collect(::updatePaymentStatus)
+            }
+
+            launch {
+                TerminalEventListener.onUnexpectedReaderDisconnect.collect {
+                    connectReader()
+                }
+            }
+        }
+    }
+
     fun discoveryReaders() {
-        discoveryTask = Terminal.getInstance().discoverReaders(
-            config,
-            discoveryListener,
-            discoveryCallback
-        )
+        try {
+            discoveryTask = Terminal.getInstance().discoverReaders(
+                config,
+                discoveryListener,
+                discoveryCallback
+            )
+        } catch (e: SecurityException) {
+            Log.e(Config.TAG, "Failed to discover readers", e)
+        }
```

Comment:
What is throwing the `SecurityException`?

Reply:
Ah I think it would be a missing location check in this particular case. Can you use a suppression annotation on the method/callsite instead?

```kotlin
@Suppress("MissingPermission")
```

Reply:
Yeah, will do